boot/boot_serial/src/boot_serial.c

```diff
@@ -439,6 +439,45 @@ bs_upload(char *buf, int len)
     flash_area_close(fap);
 }
 
+#ifdef MCUBOOT_BOOT_MGMT_ECHO
+static bool 
+decode_echo(cbor_state_t *state, cbor_string_type_t *result)
+{
+    size_t bsstrdecoded;
+
+    if (!map_start_decode(state)) {
+        return false;
+    }
+    if (!multi_decode(2, 2, &bsstrdecoded, (void *)tstrx_decode, state, result, sizeof(cbor_string_type_t))) {
+        return false;
+    }
+    map_end_decode(state);
+    return true;
+}
+
+
+static void
+bs_echo(char *buf, int len)
+{
+    const char *echo_buf="";
+    size_t echo_buflen = strlen(echo_buf);
+    size_t bsstrdecoded;
+
+    cbor_string_type_t str[2];
+
+    if (entry_function((const uint8_t *)buf, len, (const void *)str, &bsstrdecoded, (void *)decode_echo, 1, 2)) {
```

Comment:
what is the point of so many casts?

Reply:
I took the cbor_decode_Upload  as example. I could remove the str cast, but all others are needed, but harmless. I'm not really into cbor & all of it's functionality, so I just took what I could find in the current (generated) code. Suggestions or pointers to look into are welcome to optimize of course.